### ISSUE TITLE
Support for 1.16

### ISSUE THREAD

**Issue by dmulloy2** [dmulloy2/ProtocolLib#880]
General roadmap:
- [x] Waiting on Spigot
- [x] Minimum viable support
  - [x] Update version info and constants
  - [x] Generate new packet list
  - [x] Update PL wrappers
  - [x] Update unit tests
  - [x] Test
- [x] Full support
  - [x] Update attributes
  - [x] Fix tile entity accessors
  - [x] Some kind of workaround for sending chat packets with default UUID
  - [x] Update dimensions
- [x] Create new wrappers - as requested
- [ ] Update Packet Wrapper - after the dust settles

Notable packet changes
- Dimension manager wrapped in a resource key
- UUID added to chat packet
  - Will throw an internal error if left out
  - Workaround: `packet.getUUIDs().writeSafely(0, new UUID(0L, 0L))`

**Comment by dmulloy2** [comment]
https://ci.dmulloy2.net/job/ProtocolLib/lastSuccessfulBuild/

**Comment by xph11** [comment]
Getting this error spammed in console: https://pastebin.com/CfWpBD3v

Latest paper spigot 1.16.1
Latest ProtocolLib dev build (the one from few mins ago with initial support for 1.16.1)

**Comment by dmulloy2** [comment]
@xph11 that's a known issue with chat packets. Looking into a way to fix it PL side but for now the workaround is for plugins that send chat packets to do this:
`packet.getUUIDs().writeSafely(0, new UUID(0L, 0L))`

**Comment by xph11** [comment]
@dmulloy2 I am a server owner, not a developer. Just thought I should report this error, though I didn't realize it has to do with another plugin and not ProtocolLib itself. Sorry :)

**Comment by Ste3et** [comment]
Dear @dmulloy2 there is huge change at https://wiki.vg/Pre-release_protocol#Entity_Equipment
Here is a look of the NMS PacketPlayOutEntityEquipment:
https://cdn.discordapp.com/attachments/601863962904100866/726943300065689680/unknown.png
I hope You as reflection King can find a working solution to use the packet, i try it with reflection but these generic objectlist @.@ waste 4 houres of my life without solution

**Comment by jiangdashao** [comment]
Error when switching hands:
`java.lang.IllegalArgumentException: No enum constant com.comphenix.protocol.wrappers.EnumWrappers.PlayerDigType.SWAP_ITEM_WITH_OFFHAND`

1.16 changed PlayerDigType:
![image](https://user-images.githubusercontent.com/21152113/86031991-3d3d9580-ba69-11ea-97dd-fe857bb8d383.png)
this is 1.15:
![image](https://user-images.githubusercontent.com/21152113/86032023-49295780-ba69-11ea-86bd-429029902b37.png)

**Comment by dmulloy2** [comment]
@Ste3et yeah unfortunately that's a real pain to do reflectively. Doable, but annoying that they couldn't just use a map. I'll look into a good solution PL side

**Comment by Ste3et** [comment]
Here is my try to reflect the Pair<F, S> from com.mojang.datafixers.util
https://gist.github.com/Ste3et/834fe69712e5e6c7ea7b243de3213538
But how i can add it to the Wrapper i didn't know 😅

**Comment by Brokkonaut** [comment]
I made this class: https://pastebin.com/DjP9GtmB (see at the end for the usage)
Tested on recent Paper 1.16.1

**Comment by dmulloy2** [comment]
@Ste3et @Brokkonaut Added official support in https://github.com/dmulloy2/ProtocolLib/commit/f19bfc613e589ead899891e4100b77de2100f76e

API is `packet.getSlotStackPairLists()`

**Comment by Ste3et** [comment]
Thank you @dmulloy2 for spend your time it works great

**Comment by juliarn** [comment]
Hey, I tried using the ``getSlotStackPairLists()``-method by adding the 4.6.0-SNAPSHOT dependency to my pom (using the https://repo.dmulloy2.net/nexus/repository/public/ repo), but I can't seem to find the method. 
Is the build not contained in the repo yet?

**Comment by ShayBox** [comment]
Here's something else broken with 1.16 https://paste.nomsy.net/raw/abapulogos

**Comment by patrick-mc** [comment]
> Hey, I tried using the `getSlotStackPairLists()`-method by adding the 4.6.0-SNAPSHOT dependency to my pom (using the https://repo.dmulloy2.net/nexus/repository/public/ repo), but I can't seem to find the method.
> Is the build not contained in the repo yet?

@dmulloy2 for the ease of maven/gradle usage, could you add the lastest build to the nexus repository? Thanks

**Comment by Rincewind34** [comment]
WrappedChatComponent#fromText doesn't support the new hex color notation (§x§9§9§9§9§9§9)

**Comment by dmulloy2** [comment]
@Rincewind34 can you make a separate issue for that?

**Comment by soshimee** [comment]
1.16.4 broke it.

**Comment by TheGam1ngWizard** [comment]
Hiya, there's an issue with your Jenkins site.

I can't get the current protocollib
https://ci.dmulloy2.net/job/ProtocolLib/lastSuccessfulBuild/

**Comment by leeeddin** [comment]
update PacketWrapper please
